f.go

```diff
@@ -0,0 +1,199 @@
+// Copyright 2016 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+//
+// Modified by KEEP SEZC to expose F compression function.
```

Comment:
Going to touch this up after this PR

Reply:
Yeah, I had no idea what to put here. Worth double-checking if we do not have to alter `LICENSE` as well.